src/lhttpc_client.erl

```diff
@@ -150,6 +150,9 @@ execute(From, Host, Port, Ssl, Path, Method, Hdrs0, Body, Options) ->
     %% Get a socket for the pool or exit
     %Socket = lhttpc_manager:ensure_call(Pool, SocketRequest, Options),
     Socket = lhttpc_manager:ensure_call(Pool, self(), Host, Port, Ssl, Options),
+    % get the socket connect settings from client or use defaults
+    DefConOptions = application:get_env(lhttpc, connect_options, []),
+    DefConTimeout = proplists:get_value(DefConOptions, connect_timeout, infinity),
```

Comment:
This looks backwards. Please change to `proplists:get_value(connect_timeout, DefConOptions, infinity)`.